python/src/tasks/differential_expression.py

```diff
@@ -25,13 +26,39 @@ def _format_result(self, result):
         # Return a list of formatted results.
         return [Result(result)]
 
+    # Marks with a special string the cells that are not included in the experiment
+    def mark_cells_not_in_basis_set(self, raw_adata, resp):
+        basis_cell_set_name = self.task_def.get("basis")
+
+        # Ignore basis if it is all cells in the experiment or not specified (default behaviour for previous versions)
+        if not basis_cell_set_name or ("all" in basis_cell_set_name.lower()):
+            return set()
+
+        basis_cell_set = set(find_cells_by_set_id(basis_cell_set_name, resp))
+
+        raw_adata.obs["condition"].loc[
+            ~raw_adata.obs["cell_ids"].isin(basis_cell_set)
+        ] = "FilteredOut"
```

Comment:
what does this piece of code do?

Reply:
it is the special string that is used to mark cells not included in the experiment, I'll add a comment explaining that particular part of the function.